pgmpy/models/DoubleMLRegressor.py

```diff
@@ -0,0 +1,188 @@
+"""flake8: noqa"""
+
+from typing import Any, List, Optional, Sequence, Tuple
+
+import numpy as np
+import pandas as pd
+from sklearn.base import BaseEstimator, RegressorMixin, clone
+from sklearn.utils.validation import check_is_fitted, check_X_y
+
+
+class DoubleMLRegressor(BaseEstimator, RegressorMixin):
+    def __init__(
+        self,
+        dag: Any,
+        estimator_g: Any,
+        estimator_m: Optional[Any] = None,
+        adjustment_set: Optional[Sequence[str]] = None,
+        treatment_col: Optional[str] = None,
+        n_folds: int = 5,
+        random_state: Optional[int] = None,
+    ):
+        self.dag = dag
+        self.estimator_g = estimator_g
+        self.estimator_m = estimator_m
+        self.adjustment_set = adjustment_set
+        self.treatment_col = treatment_col
+        self.n_folds = n_folds
+        self.random_state = random_state
+
+    def _more_tags(self):
+        return {"requires_y": True, "no_sparse_input": True}
+
+    # ---------- small helpers ----------
+    def _as_dataframe(self, X) -> pd.DataFrame:
+        if isinstance(X, pd.DataFrame):
+            return X.copy()
+        arr = np.asarray(X)
+        if arr.ndim == 1:
+            arr = arr.reshape(-1, 1)
+        cols = [f"x{i}" for i in range(arr.shape[1])]
+        return pd.DataFrame(arr, columns=cols)
+
+    def _resolve_treatment_and_adjustments(
+        self, df: pd.DataFrame
+    ) -> Tuple[str, List[str]]:
+        # treatment priority: explicit -> dag.exposure -> first column
+        if self.treatment_col is not None and self.treatment_col in df.columns:
+            treat = self.treatment_col
+        else:
+            exp = getattr(self.dag, "exposure", None)
+            if exp is None:
+                treat = df.columns[0]
+            else:
+                exp_list = list(exp) if isinstance(exp, (list, tuple, set)) else [exp]
+                if len(exp_list) != 1:
+                    raise NotImplementedError(
+                        "Only single exposure supported in " "skeleton."
+                    )
+                treat = exp_list[0] if exp_list[0] in df.columns else df.columns[0]
+
+        # adjustments priority: explicit -> dag.get_adjustment_set() -> []
+        if self.adjustment_set is not None:
+            adj = [c for c in self.adjustment_set if c in df.columns and c != treat]
+        else:
+            adj = []
+            if hasattr(self.dag, "get_adjustment_set"):
+                try:
```

Comment:
Assume that the adjustment set is already present in the model. If not, throw an error.